splinter_ros/CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+cmake_minimum_required(VERSION 3.8)
+project(splinter_ros)
+
+find_package(ament_cmake REQUIRED)
+find_package(splinter_vendor REQUIRED)
+
+add_library(splinter1d SHARED src/splinter1d.cpp)
+target_link_libraries(splinter1d PRIVATE splinter)
+target_include_directories(splinter1d PUBLIC
+  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+  "$<INSTALL_INTERFACE:include>"
+)
+
+add_library(splinter2d SHARED src/splinter2d.cpp)
+target_link_libraries(splinter2d PRIVATE splinter)
+target_include_directories(splinter2d PUBLIC
+  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+  "$<INSTALL_INTERFACE:include>"
+)
+
+install(TARGETS splinter1d splinter2d
+  EXPORT "export_splinter_ros"
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION lib
+  INCLUDES DESTINATION include
+)
+
+install(DIRECTORY include
```

Comment:
I've never liked this semantic, but need a leading forward slash on 
`install(DIRECTORY include/`
or the result will be `install/splinter_ros/include/include/splinter_ros`